File: codeforces_code/ECR168/C.py
for _ in range(int(input())):
    n = int(input())
    s = list(input())
    i, j= 0, 1
    ans = 0
    while i < n and j < n:
        if j <= i:
            j += 1
        elif (s[i] == '_' or s[i] == '(') and (s[j] == ')' or s[j] == '_'):
            ans += j - i
            s[i] = '('
            s[j] = ')'
            i += 1
            j += 1
        elif s[i] == '_' or s[i] == '(':
            j += 1
        else:
            i += 1
    print(ans)

File: codeforces_code/ECR168/C.cpp
#include <bits/stdc++.h>
using namespace std;

void solve()
{
    int n;
    cin >> n;
    string s;
    cin >> s;
    int i = 0, j = 1;
    int ans = 0;
    while (i < n && j < n)
    {
        cout << i << ' ' << j;
        while (i < n && (s[i] != '_' || s[i] != '('))
            i++;
        while (j < n && (s[j] != ')' || s[j] != '_'))
            j++;
        ans += (j - i);
        s[i] = '(', s[j] = ')';
        cout << s << '\n';
        i++, j++;
    }
    cout << ans << '\n';
}

int main()
{
    int tt;
    cin >> tt;
    while (tt--)
        solve();

    return 0;
}